core/src/main/scala/io/finch/request/package.scala

```diff
@@ -162,6 +180,8 @@ package object request {
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) =
         try number.toFuture
         catch { case _: IllegalArgumentException => ValidationFailed(param, rule).toFutureException }
+
+      override def toString: String = s"param: $param"
```

Comment:
What I personally like about `which` is that it fits into the whole story pretty well:

```scala
RequiredIntParam("a") which ShouldBePositive // vs
RequiredIntParam("") should BePositive
```

1. It feels like a natural english `required int param which should be positive`
2. In case of `which` we might define the bunch of rules that all start with `Should`, so users will find them easily just by autocompletion "Should...". In case of `should` method we would have both `BePositive` and `NotBePositive`.

I would stick with `which` and rules started with `Should`.